optimizer_wrapper.rb

```diff
@@ -363,78 +374,84 @@ def self.split_independent_vrp_by_skills(vrp)
-      sub_service_vrp.resolution_iterations_without_improvment = vrp.resolution_iterations_without_improvment&.*(split_ratio)&.ceil
+      this_sub_size = vrp.services.all?{ |service| service.sticky_vehicles.any? } ? sub_vrp.vehicles.size :
+        sub_vrp.services.size * [1, sub_vrp.vehicles.size].min
+      adjust_independent_duration(sub_vrp, this_sub_size, total_size)
     }
 
-    return independent_vrps if unused_vehicles_indices.empty?
+    return independent_vrps if unused_vehicle_indices.empty?
 
-    sub_service_vrp = Interpreters::SplitClustering.build_partial_service_vrp({ vrp: vrp }, [], unused_vehicles_indices)
+    sub_service_vrp = Interpreters::SplitClustering.build_partial_service_vrp({ vrp: vrp }, [], unused_vehicle_indices)
     sub_service_vrp[:vrp].matrices = []
     independent_vrps.push(sub_service_vrp[:vrp])
 
     independent_vrps
   end
 
-  def self.split_independent_vrp_by_sticky_vehicle(vrp)
-    vrp.vehicles.map.with_index{ |vehicle, v_i|
-      vehicle_id = vehicle.id
-      service_ids = vrp.services.select{ |s| s.sticky_vehicles.map(&:id) == [vehicle_id] }.map(&:id)
-      service_vrp = {
-        service: nil,
-        vrp: vrp,
-      }
-      sub_service_vrp = Interpreters::SplitClustering.build_partial_service_vrp(service_vrp, service_ids, [v_i])
-      split_ratio = 1.0 / vrp.vehicles.size
-      sub_service_vrp[:vrp].resolution_duration = vrp.resolution_duration&.*(split_ratio)&.ceil
-      sub_service_vrp[:vrp].resolution_minimum_duration = (vrp.resolution_minimum_duration || vrp.resolution_initial_time_out)&.*(split_ratio)&.ceil
-      sub_service_vrp[:vrp].resolution_iterations_without_improvment = vrp.resolution_iterations_without_improvment&.*(split_ratio)&.ceil
-      sub_service_vrp[:vrp]
-    }
+  def self.adjust_independent_duration(vrp, this_sub_size, total_size)
+    split_ratio = this_sub_size.to_f / total_size
+    vrp.resolution_duration = vrp.resolution_duration&.*(split_ratio)&.ceil
+    vrp.resolution_minimum_duration =
+      vrp.resolution_minimum_duration&.*(split_ratio)&.ceil
+    vrp.resolution_iterations_without_improvment =
+      vrp.resolution_iterations_without_improvment&.*(split_ratio)&.ceil
+    vrp
   end
 
   def self.split_independent_vrp(vrp)
     # Don't split vrp if
-    return [vrp] if (vrp.vehicles.size <= 1) || vrp.services.empty? # there might be zero services
+    # - No vehicle
+    # - No service
+    # - there is multimodal subtours
+    return [vrp] if (vrp.vehicles.size <= 1) || vrp.services.empty? || vrp.subtours&.any? # there might be zero services
 
-    if vrp.services.all?{ |s| s.sticky_vehicles.size == 1 }
-      return split_independent_vrp_by_sticky_vehicle(vrp)
-    end
-    # Cannot split if there is multimodal subtours
-    if !vrp.subtours&.any? && vrp.services.all?{ |s| s.sticky_vehicles.empty? }
-      return split_independent_vrp_by_skills(vrp)
-    end
+    # - a service with no service (or sticky vehicle see: expand_data.rb:sticky_as_skills)
```

Comment:
This is not very important but I do not understand this comment, could you eventually rephrase so it is more explicit for all ?